### ISSUE TITLE
Load environment variables from .env file and global variables

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#10928]
Refs: #10359

- [x] anyOS @IanMatthewHuff 
- [x] anyOS @rchiodo

Complexity: 3

Authors: @DonJayamanne 

[Create Issue](https://github.com/microsoft/vscode-jupyter/issues/new?body=Testing+%237538%0A%0A&assignees=DonJayamanne)

[Create Issue](https://github.com/microsoft/vscode-jupyter/issues/new?body=Testing+%2310928%0A%0A&assignees=DonJayamanne)

---

### Requirements

- Install prerelease jupyter extension into VS Code Insiders (desktop)
- Have a python environment that you can run jupyter from. (https://jupyter.org/install)
- Be familiar with jupyter notebooks
- Use Python 3.6 or later (please ensure you aren't using Python 2, use `python --version` to check the version)

### Test loading variables from .env file

1. Open a folder in VS Code
2. Add a file named `.env`
3. Add an environment variable to the file
4. Create an ipynb file in the root directory
5. Run a cell in the ipynb file to print the value of the above env variable
```python
import os
print(os.getenv(<env variable>))
```
7. Verify the right value is printed
8. Updated the value in the .env file
9. Restart the kernel and verify the new value is printed.

### Test loading variables from global env variables
1. Exit VS Code completely
1. Create a global environment variable (on windows use env varaibles editor and unix update ~/.bashrc or the like)
2. Open a folder in VS Code
5. Create an ipynb file in the root directory
6. Run a cell in the ipynb file to print the value of the above env variable
```python
import os
print(os.getenv(<env variable>))
```
7. Verify the right value is printed
8. Updated the environment variable
9. Exit vscode completely and reload VS Code
10. verify the new value is printed when running a cell again

**Comment by IanMatthewHuff** [comment]
Looks good. All the scenarios work for me. I think Rich is on Windows, which is good since I'm pretty sure at least all of the .env file stuff already worked for me on Mac.

**Comment by rchiodo** [comment]
I'm having trouble with ZMQ disabled. I can't run the IW at the moment.

**Comment by rchiodo** [comment]
I'm blocked on testing this because non ZMQ doesn't work for me.

Something submitted on this day broke non ZMQ for me:
https://dev.azure.com/monacotools/Monaco/_traceability/runview/changes?currentRunId=178730

**Comment by rchiodo** [comment]
Entered #10940

**Comment by rchiodo** [comment]
Works now after the ZMQ fix

**Comment by jtele2** [comment]
I am trying to get this to work. Does the file have to be the `.ipynb` extenstion? Does it have to be in the root folder? 

I'm using the Python Interactive with regular `.py` files and the workflow proposed by @DonJayamanne is not working - although it is exactly what I am looking to do.

**Comment by DonJayamanne** [comment]
@jtele2 
please file a separate issue with the results and steps required to repro the issue
